pkg/util/hostname.go

```diff
@@ -249,12 +274,21 @@ func GetHostname() (string, error) {
 		err = nil
 	}
 
-	cache.Cache.Set(cacheHostnameKey, hostName, cache.NoExpiration)
-	setHostnameProvider(provider)
+	hostnameData := setHostnameData(cacheHostnameKey, hostName, provider)
 	if err != nil {
 		expErr := new(expvar.String)
 		expErr.Set(fmt.Sprintf(err.Error()))
 		hostnameErrors.Set("all", expErr)
 	}
-	return hostName, err
+	return hostnameData, err
+}
+
+func checkIfHostnameUsedAsCanonicalHostname(hostname string) bool {
```

Comment:
let's add a comment explaining what this function's purpose is

Reply:
and a more concise name could be better. Suggestion: `isHostnameCanonicalForIntake`

Reply:
Done.